### PR TITLE
upgrade polkadot 0.9.43

### PR DESCRIPTION
**Pull request by blasrodri**

### CODE REVIEW

**Review by dzmitry-lahoda — DISMISSED** [review]

**Review by RustNinja — DISMISSED** [review]

**Review by RustNinja — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
```
bash-5.1$ /home/dz/.cargo/bin/zepter  lint propagate-feature --feature runtime-benchmarks -p pallet-ibc --workspace
crate 'pallet-ibc'
  feature 'runtime-benchmarks'
    must propagate to:
      pallet-ibc-ping
Found 1 issue (run with --fix to fix).
bash-5.1$ /home/dz/.cargo/bin/zepter  lint propagate-feature --feature std  -p pallet-ibc --workspace
crate 'pallet-ibc'
  feature 'std'
    must propagate to:
      ics08-wasm
      pallet-ibc-ping
      simple-iavl
Found 3 issues (run with --fix to fix).
bash-5.1$ 
```

**Review by RustNinja — DISMISSED** [review]

**Review by RustNinja — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
```
ics10-grandpa-cw-deps> warning: build failed, waiting for other jobs to finish...
error: builder for '/nix/store/cgmjbzhxj1gjmq5g6778vc47r67x91c5-ics10-grandpa-cw-deps-0.1.drv' failed with exit code 101;
       last 10 log lines:
       >    Compiling fs-err v2.9.0
       > error[E0433]: failed to resolve: use of undeclared crate or module `imp`
       >    --> /nix/store/s61z89zhi2j60pij31zlbfbz1lszvlz1-vendor-cargo-deps/c19b7c6f923b580ac259164a89f2577984ad5ab09ee9d583b888f934adbbe8d0/getrandom-0.2.8/src/lib.rs:290:5
       >     |
       > 290 |     imp::getrandom_inner(dest)
       >     |     ^^^ use of undeclared crate or module `imp`
       >
       > For more information about this error, try `rustc --explain E0433`.
       > error: could not compile `getrandom` (lib) due to 2 previous errors
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/cgmjbzhxj1gjmq5g6778vc47r67x91c5-ics10-grandpa-cw-deps-0.1.drv'.
error: 1 dependencies of derivation '/nix/store/3js0q09iw66fiyzl25d1jdnl1vcdvyij-ics10-grandpa-cw-0.1.drv' failed to build
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'docker-image-devnet-xc.tar.gz'
         whose name attribute is located at /nix/store/sql9ggq8nr9w8z7s71mqp8jdzvizhc1i-source/pkgs/stdenv/generic/make-derivation.nix:300:7

       … while evaluating attribute 'baseJson' of derivation 'docker-image-devnet-xc.tar.gz'

         at /nix/store/sql9ggq8nr9w8z7s71mqp8jdzvizhc1i-source/pkgs/build-support/docker/default.nix:598:18:

          597|           imageTag = if tag == null then "" else tag;
          598|           inherit fromImage baseJson;
             |                  ^
          599|           layerClosure = writeReferencesToFile layer;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: 1 dependencies of derivation '/nix/store/8n4y8zzgjh6gm72yn4789a3nwv9rm24f-ics10-grandpa-cw.drv' failed to build
Stopped watching /nix/store and waiting for queue to empty ...
Done.
Pushing /nix/store/41bgp64nblnnzqvv1zv1m5sl3544xw9i-source (126.60 MiB)
will try execute 'cachix watch-exec --compression-level 16 --compression-method zstd --jobs 16 composable nix -- build .#devnet-xc-image --no-update-lock-file --accept-flake-config  -L 2> >(tee --append  >&2)'
Error: Process completed with exit code 1.
```
something changed regarding STD feature in light client building.

**Comment by dzmitry-lahoda** [comment]
here they hidden https://github.com/paritytech/zombienet/blob/main/docs/src/network-definition-spec.md

**Comment by dzmitry-lahoda** [comment]
For upgradeability we all runt wasm runtimes, so to have proper gas we all run benchmarks. Currently benchmarks on wasm of pallet-ibc are broken

```
bash-5.1$ cargo check --no-default-features --target wasm32-unknown-unknown --package pallet-ibc --features runtime-benchmarks
warning: /home/dz/github.com/ComposableFi/centauri/utils/subxt/codegen/Cargo.toml: unused manifest key: dependencies.wasm-testbed.release
    Checking getrandom v0.2.8
   Compiling quote v1.0.33
   Compiling generic-array v0.14.6
   Compiling num-traits v0.2.15
   Compiling typenum v1.16.0
   Compiling ahash v0.7.6
   Compiling subtle v2.4.1
   Compiling regex-syntax v0.6.28
   Compiling proc-macro-crate v1.3.1
   Compiling memchr v2.5.0
   Compiling paste v1.0.11
   Compiling indexmap v1.9.2
   Compiling ahash v0.8.3
    Checking log v0.4.20
    Checking pin-project-lite v0.2.13
   Compiling anyhow v1.0.69
   Compiling wasm-bindgen-shared v0.2.84
error: the wasm32-unknown-unknown target is not supported by default, you may need to enable the "js" feature. For more information see: https://docs.rs/getrandom/#webassembly-support
   --> /home/dz/.cargo/registry/src/index.crates.io-6f17d22bba15001f/getrandom-0.2.8/src/lib.rs:263:9
    |
263 | /         compile_error!("the wasm32-unknown-unknown target is not supported by \
264 | |                         default, you may need to enable the \"js\" feature. \
265 | |                         For more information see: \
266 | |                         https://docs.rs/getrandom/#webassembly-support");
    | |________________________________________________________________________^

    Checking bitvec v1.0.1
   Compiling convert_case v0.4.0
    Checking tinyvec_macros v0.1.1
   Compiling futures-core v0.3.28
error[E0433]: failed to resolve: use of undeclared crate or module `imp`
   --> /home/dz/.cargo/registry/src/index.crates.io-6f17d22bba15001f/getrandom-0.2.8/src/lib.rs:290:5
    |
290 |     imp::getrandom_inner(dest)
    |     ^^^ use of undeclared crate or module `imp`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `getrandom` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
bash-5.1$ ^C
```

**Comment by dzmitry-lahoda** [comment]
locally runs 
image

**Comment by dzmitry-lahoda** [comment]
image

locally runs perfectly.

**Comment by dzmitry-lahoda** [comment]
i do not have access to gce machine - i cannot tell what is wrong here.

**Comment by dzmitry-lahoda** [comment]
CI is not blocker. things run locally, it can be fixed locally I think.

**Review by dzmitry-lahoda — DISMISSED** [review]

File: light-clients/common/src/lib.rs [hunk 1]
```diff
@@ -194,8 +194,9 @@ pub fn decode_timestamp_extrinsic(ext: &Vec<u8>) -> Result<u64, anyhow::Error> {
 	// https://github.com/paritytech/substrate/blob/d602397a0bbb24b5d627795b797259a44a5e29e9/primitives/trie/src/lib.rs#L99-L101
 	// Decoding from the [2..] because the timestamp inmherent has two extra bytes before the call
 	// that represents the call length and the extrinsic version.
-	let (_, _, timestamp): (u8, u8, Compact<u64>) = codec::Decode::decode(&mut &ext[2..])
-		.map_err(|err| anyhow!("Failed to decode extrinsic: {err}"))?;
+	let (_, _, timestamp): (u8, u8, Compact<u64>) =
+		codec::Decode::decode(&mut &ext[ext.len() - 7..])
```
Comment (RustNinja) [inline light-clients/common/src/lib.rs:198]: just checked that it is fine to use &ext[5..]
and need to decode directly into Compact<u64> and not (u8, u8, Compact<u64>)

so it supposed to be like this

```
let ret: Compact<u64> = codec::Decode::decode(&mut &ext[5..])
		.map_err(|err| anyhow!("Failed to decode extrinsic: {err}"))?;
```

**Review by RustNinja — DISMISSED** [review]

**Comment by RustNinja** [comment]
error: toolchain 'nightly-x86_64-unknown-linux-gnu' is not installed @dzmitry-lahoda

**Comment by dzmitry-lahoda** [comment]
@RustNinja what command you run to get error?

**Comment by dzmitry-lahoda** [comment]
why CI does not runs jobs?

**Pull request closed without merging**